app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js

```diff
@@ -156,46 +155,34 @@ export function validateAddEthereumChainParams(params, end) {
 export async function switchChain(
   res,
   end,
-  origin,
   chainId,
-  requestData,
   networkClientId,
   approvalFlowId,
   {
     isAddFlow,
-    getChainPermissionsFeatureFlag,
     setActiveNetwork,
     endApprovalFlow,
-    requestUserApproval,
     getCaveat,
     requestPermittedChainsPermission,
     grantPermittedChainsPermissionIncremental,
   },
 ) {
   try {
-    if (getChainPermissionsFeatureFlag()) {
-      const { value: permissionedChainIds } =
-        getCaveat({
-          target: PermissionNames.permittedChains,
-          caveatType: CaveatTypes.restrictNetworkSwitching,
-        }) ?? {};
-
-      if (
-        permissionedChainIds === undefined ||
-        !permissionedChainIds.includes(chainId)
-      ) {
-        if (isAddFlow) {
-          await grantPermittedChainsPermissionIncremental([chainId]);
-        } else {
-          await requestPermittedChainsPermission([chainId]);
-        }
+    const { value: permissionedChainIds } =
+      getCaveat({
+        target: PermissionNames.permittedChains,
+        caveatType: CaveatTypes.restrictNetworkSwitching,
+      }) ?? {};
+
+    if (
+      permissionedChainIds === undefined ||
+      !permissionedChainIds.includes(chainId)
+    ) {
+      if (isAddFlow) {
+        await grantPermittedChainsPermissionIncremental([chainId]);
+      } else {
+        await requestPermittedChainsPermission([chainId]);
       }
-    } else {
-      await requestUserApproval({
-        origin,
-        type: ApprovalType.SwitchEthereumChain,
-        requestData,
-      });
```

Comment:
@adonesky1 we don't need to keep this around for snaps, right?

Reply:
Nope. Snaps don't call `wallet_switchEthereumChain`. And once they do we will be granting them permissions on each request